### PR TITLE
docs: add ADC, DAC, qencoder and PWM API description

### PR DESCRIPTION
**Pull request by michallenc**
## Summary
Adds API description of ADC, DAC, qencoder and PWM peripherals.

## Impact
Docs only.

### CODE REVIEW

File: Documentation/components/drivers/character/analog/dac/index.rst [hunk 1]
```diff
@@ -0,0 +1,75 @@
+===========
+DAC Drivers
+===========
+
+-  ``include/nuttx/analog/dac.h``. All structures and APIs needed
+   to work with DAtC drivers are provided in this header file. This
```
Comment (acassis) [inline Documentation/components/drivers/character/analog/dac/index.rst:6]:
```suggestion
   to work with DAC drivers are provided in this header file. This

File: Documentation/components/drivers/character/analog/dac/index.rst [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+The first necessary thing to be done in order to use the DAC driver from an
+application is to include the correct header filer. It contains the
+Application Programming Interface to the PWM driver. To do so, include
+
+.. code-block:: c
+
+  #include <nuttx/analog/dac.h>
+
+DAC driver is registered as a POSIX character device driver into ``/dev``
```
Comment (acassis) [inline Documentation/components/drivers/character/analog/dac/index.rst:30]:
```suggestion
DAC driver is registered as a POSIX character device file into ``/dev``

File: Documentation/components/drivers/character/analog/dac/index.rst [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    int32_t      am_data;
+  } end_packed_struct;
+
+Application Example
+~~~~~~~~~~~~~~~~~~~
+
+An example application can be found in ``nuttx-apps`` repository under
+path ``examples/dac``. It provides command line interface to write data
+to DAC channels.s
+
```
Comment (acassis) [inline Documentation/components/drivers/character/analog/dac/index.rst:54]: Please add the nsh dac command in the terminal here

File: Documentation/components/drivers/character/quadrature.rst [hunk 4]
```diff
@@ -28,3 +28,86 @@ following locations:
...
+
+The first thing to be done in order to use the quadrature encoder driver
+from an application is to include the correct header filer. It contains the
+Application Programming Interface to the driver. To do so, include
+
+.. code-block:: c
+
+  #include <nuttx/sensors/qencoder.h>
+
+Quadrature encoder driver is registered as a POSIX character device driver
```
Comment (acassis) [inline Documentation/components/drivers/character/quadrature.rst:42]:
```suggestion
Quadrature encoder driver is registered as a POSIX character device file

File: Documentation/components/drivers/character/quadrature.rst [hunk 5]
```diff
@@ -28,3 +28,86 @@ following locations:
...
+The pointer to this structure is used as an argument to ``QEIOC_GETINDEX``
+ioctl command. It gets the current encoder position, the last position of
+index and the number of index occurances.
+
+Application Example
+~~~~~~~~~~~~~~~~~~~
+
+An example application can be found in ``nuttx-apps`` repository under
+path ``examples/qencoder``. It demonstrates the basic data read
+from an encoder device.
```
Comment (acassis) [inline Documentation/components/drivers/character/quadrature.rst:103]:
```suggestion
from an encoder device. Example:

.. code-block:: console

    nsh> qe
    1.  0
    2.  0
    3.  0
    4.  1
    5.  1
    6.  1
    7.  2
    8.  2
    9.  3

File: Documentation/components/drivers/character/timers/pwm.rst [hunk 6]
```diff
@@ -32,3 +32,171 @@ Files supporting PWM can be found in the following locations:
...
+
+The first necessary thing to be done in order to use the PWM driver in an
+application is to include the header file for the NuttX timer driver. It contains
+the Application Level Interface to the PWM driver. To do so, include:
+
+.. code-block:: c
+
+  #include <nuttx/timers/pwm.h>
+
+PWM driver is registered as a POSIX character device into ``/dev`` namespace.
```
Comment (acassis) [inline Documentation/components/drivers/character/timers/pwm.rst:47]:
```suggestion
PWM driver is registered as a POSIX character device file into ``/dev`` namespace.

**Comment by acassis** [comment]
Amazing work @michallenc very useful and very well done! Kudos!!!

**Comment by michallenc** [comment]
> Amazing work @michallenc very useful and very well done! Kudos!!!

Thanks Alan! I have updated the documentation with your suggestions.

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by anchao — APPROVED** [review]

**Pull request merged**